File: backend/app/models/attendance.py
from pydantic import BaseModel, Field
from datetime import datetime


class AttendanceCreate(BaseModel):
    employee_id: str = Field(..., example="EMP001")
    date: datetime = Field(..., example="2024-06-01T09:00:00Z")
    status: str = Field(..., example="PRESENT")

File: frontend/src/types/attendance.ts
export type AttendanceStatus = "PRESENT" | "ABSENT";

export interface Attendance {
  id: string;
  employee_id: string;
  date: string;
  status: AttendanceStatus;
}

export interface AttendanceCreate {
  employee_id: string;
  date: string;
  status: AttendanceStatus;
}
